python/taichi/lang/__init__.py

```diff
@@ -456,6 +523,26 @@ def init(arch=None,
             * ``print_ir`` (bool): Prints the CHI IR of the Taichi kernels.
             * ``packed`` (bool): Enables the packed memory layout. See https://docs.taichi.graphics/lang/articles/advanced/layout.
     """
+    # Check version for users every 7 days.
+    os.makedirs(_ti_core.get_repo_dir(), exist_ok=True)
+    timestamp_path = os.path.join(_ti_core.get_repo_dir(), 'timestamp')
+    cur_date = datetime.date.today()
+    if os.path.exists(timestamp_path):
+        last_time = ''
+        with open(timestamp_path, 'r') as f:
+            last_time = f.readlines()[0].rstrip()
+        if cur_date.strftime('%Y-%m-%d') > last_time:
+            check_version()
+            with open(timestamp_path, 'w') as f:
+                f.write((cur_date +
+                         datetime.timedelta(days=14)).strftime('%Y-%m-%d'))
+                f.truncate()
+    else:
+        check_version()
+        with open(timestamp_path, 'w') as f:
```

Comment:
```suggestion
        with open(timestamp_path, 'wb') as f:
```
maybe use binary mode? (if so, same for all R/W...)

Reply:
Is binary mode better?